pkg/backup/backupschedule/backup_schedule_manager.go

```diff
@@ -332,12 +333,33 @@ func (bm *backupScheduleManager) backupGCByMaxBackups(bs *v1alpha1.BackupSchedul
 		klog.Infof("backup schedule %s/%s gc backup %s success", ns, bsName, backup.GetName())
 	}
 
-	if deleteCount == len(backupsList) {
+	if deleteCount == len(backupsList) && bm.needResetLastBackup(bs, backupsList) {
```

Comment:
Why we should check the last backup here?
If all the backups have been cleaned, we should reset the backupSchedule here.
The fix in this PR does not align with the issue described in https://github.com/pingcap/tidb-operator/issues/4220, in #4220, the cleanup behavior is good, so the GC behavior is good, right?
The issue occurs when we unpause the backupSchedule, right?
If yes, we have to see that if it is caused by the stale cache in the controller (not synced from kube-apiserver in time), if yes, what we can do to avoid or fix this.

Reply:
the problem is when deleteCount == len(backupsList) condition true，then status.lastBackup is nil，there are two case when deleteCount == len(backupsList) condition true:
1. as issue described
2. the first backup of the backupscheduler. 
the problem can easily be reoccurred on a normal k8s enviarenment as the test log described.

Reply:
the problem is caused by the stale cache in the controller as you  describe above，and i find the re-check method to make sure it in other part like this:
https://github.com/pingcap/tidb-operator/blob/b953252754b0ecd4d6041708f8c3f0574eb4484a/pkg/manager/member/orphan_pods_cleaner.go#L132